src/Service/PatternPluginManagerFactory.php

```diff
@@ -10,17 +10,10 @@
 
 use Interop\Container\ContainerInterface;
 use Laminas\Cache\PatternPluginManager;
-use Laminas\ServiceManager\FactoryInterface;
-use Laminas\ServiceManager\ServiceLocatorInterface;
+use Laminas\ServiceManager\Factory\FactoryInterface;
```

Comment:
We don't really need to implement the `FactoryInterface` at this point.

Reply:
Considering that most of these plugin managers will be pulled from the container directly via `get()` and not `build()`, I think it might be interesting to allow passing options to the _constructor_ of the factory (which would also require implementing a `__set_state()` method for serialization purposes). This would allow setting global option when configuring the container.

Thoughts?

Reply:
Imho, there is no need to pass any options to the `PatternPluginManager`. 

I'd rather create another PR where I remove `FactoryInterface` from those services/plugin managers which do not need a factory implementing that interface.

If a project might want to add custom stuff to the `PatternPluginManager`, they can just overwrite the factory for it instead of having some "magic" in the existing factory.

Reply:
One thing to note: removing the reliance on `FactoryInterface` means you can update to typehint on PSR-11 instead of container-interop interfaces... which I think would be a good idea for this major version.

Reply:
Thats exactly what I did in boesing/laminas-cache#1
I'll merge that into this branch.